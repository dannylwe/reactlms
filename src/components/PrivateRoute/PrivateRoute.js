import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route render={props => (rest.isLoggedIn ? <Component {...props} {...rest} />
    : (
      <Redirect to={{
        pathname: '/login',
        state: { from: props.location },
      }}
      />
    ))}
  />
);
PrivateRoute.defaultProps = {
  component: '',
  location: '',
};

PrivateRoute.propTypes = {
  component: PropTypes.func,
  location: PropTypes.string,
};

export default PrivateRoute;
