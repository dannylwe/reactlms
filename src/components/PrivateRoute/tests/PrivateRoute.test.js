import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import PrivateRoute from '../PrivateRoute';

describe('private componet', () => {
  it('should render successfully', () => {
    const props = {
      isLoggedIn: true,
    };
    const wrapper = renderer.create(
      <MemoryRouter>
        <PrivateRoute isLoggedIn={props.isLoggedIn} />
      </MemoryRouter>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('should not render successfully', () => {
    const props = {
      isLoggedIn: false,
    };
    const wrapper = shallow(
      <MemoryRouter>
        <PrivateRoute isLoggedIn={props.isLoggedIn} />
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
