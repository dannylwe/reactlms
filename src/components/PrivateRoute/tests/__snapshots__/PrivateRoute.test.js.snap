// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`private componet should not render successfully 1`] = `ShallowWrapper {}`;

exports[`private componet should render successfully 1`] = `
<
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "wyap0m",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "wyap0m",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  isLoggedIn={true}
  location=""
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
/>
`;
